contrib/http_util/src/main/java/io/opencensus/contrib/http/HttpHandler.java

```diff
@@ -0,0 +1,113 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opencensus.contrib.http;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.common.annotations.VisibleForTesting;
+import io.opencensus.common.ExperimentalApi;
+import io.opencensus.contrib.http.util.HttpTraceUtil;
+import io.opencensus.trace.AttributeValue;
+import io.opencensus.trace.MessageEvent;
+import io.opencensus.trace.MessageEvent.Type;
+import io.opencensus.trace.Span;
+import javax.annotation.Nullable;
+
+/**
+ * Base class for handling request on http client and server.
+ *
+ * @param <Q> the HTTP request entity.
+ * @param <P> the HTTP response entity.
+ * @since 0.18
+ */
+@ExperimentalApi
+public abstract class HttpHandler<Q, P> {
+
+  /** The {@link HttpExtractor} used to extract information from request/response. */
+  @VisibleForTesting final HttpExtractor<Q, P> extractor;
+
+  /** Constructor to allow access from same package subclasses only. */
+  HttpHandler(HttpExtractor<Q, P> extractor) {
+    checkNotNull(extractor, "extractor");
+    this.extractor = extractor;
+  }
+
+  /**
+   * A convenience to record a {@link MessageEvent} with given parameters.
+   *
+   * @param span the span which this {@code MessageEvent} will be added to.
+   * @param id the id of the event.
+   * @param type the {@code MessageEvent.Type} of the event.
+   * @param uncompressedMessageSize size of the message before compressed (optional).
+   * @param compressedMessageSize size of the message after compressed (optional).
+   * @since 0.18
+   */
+  static void recordMessageEvent(
+      Span span, long id, Type type, long uncompressedMessageSize, long compressedMessageSize) {
+    MessageEvent messageEvent =
+        MessageEvent.builder(type, id)
+            .setUncompressedMessageSize(uncompressedMessageSize)
+            .setCompressedMessageSize(compressedMessageSize)
+            .build();
+    span.addMessageEvent(messageEvent);
+  }
+
+  /**
+   * Instrument an HTTP span after a message is sent.
+   *
+   * @param span the span.
+   * @param messageId an id for the message.
+   * @param messageSize the size of the message.
+   * @since 0.18
+   */
+  public final void handleMessageSent(Span span, long messageId, long messageSize) {
+    checkNotNull(span, "span");
+    recordMessageEvent(span, messageId, Type.SENT, messageSize, 0L);
+  }
+
+  /**
+   * Instrument an HTTP span after a message is received.
+   *
+   * @param span the span.
+   * @param messageId an id for the message.
+   * @param messageSize the size of the message.
+   * @since 0.18
+   */
+  public final void handleMessageReceived(Span span, long messageId, long messageSize) {
+    checkNotNull(span, "span");
+    // record message size
```

Comment:
This comment should be "record compressed size".  Same comment in handleMessageSent.

Reply:
done.